packages/dynamic_layouts/lib/src/wrap_layout.dart

```diff
@@ -0,0 +1,270 @@
+  ];
+
+  // In this case the function getGeometryForChildIndex only works to get
+  // the geometry for the child, where the layout actually happens is in
+  // updateGeometryForChildIndex.
+  @override
+  DynamicSliverGridGeometry getGeometryForChildIndex(int index) {
+    return DynamicSliverGridGeometry(
+      scrollOffset: 0,
+      crossAxisOffset: 0,
+      mainAxisExtent: childMainAxisExtent,
+      crossAxisExtent: childCrossAxisExtent,
+    );
+  }
+
+  @override
+  DynamicSliverGridGeometry updateGeometryForChildIndex(
+      int index, Size childSize) {
+    final double scrollOffset = model.last.scrollOffset;
+    final double currentSizeUsed = model.last.currentSizeUsed;
+    late final double addedSize;
+
+    switch (scrollDirection) {
+      case Axis.vertical:
+        addedSize = currentSizeUsed + childSize.width + crossAxisSpacing;
+        break;
+      case Axis.horizontal:
+        addedSize = currentSizeUsed + childSize.height + mainAxisSpacing;
+        break;
+    }
+
+    if (addedSize > crossAxisExtent && model.last.currentSizeUsed > 0.0) {
+      switch (scrollDirection) {
+        case Axis.vertical:
+          model.add(
+            _RunMetrics(
+              maxSliver: childSize.height + mainAxisSpacing,
+              currentSizeUsed: childSize.width + crossAxisSpacing,
+              scrollOffset:
+                  scrollOffset + model.last.maxSliver + mainAxisSpacing,
+            ),
+          );
+          break;
+        case Axis.horizontal:
+          model.add(
+            _RunMetrics(
+              maxSliver: childSize.width + crossAxisSpacing,
+              currentSizeUsed: childSize.height + mainAxisSpacing,
+              scrollOffset:
+                  scrollOffset + model.last.maxSliver + crossAxisSpacing,
+            ),
+          );
+          break;
+      }
+
+      return DynamicSliverGridGeometry(
+        scrollOffset: model.last.scrollOffset,
+        crossAxisOffset: 0.0,
+        mainAxisExtent: childSize.height + mainAxisSpacing,
+        crossAxisExtent: childSize.width + crossAxisSpacing,
+      );
+    } else {
+      model.last.currentSizeUsed = addedSize;
+    }
+
+    switch (scrollDirection) {
+      case Axis.vertical:
+        if (childSize.height + mainAxisSpacing > model.last.maxSliver) {
+          model.last.maxSliver = childSize.height + mainAxisSpacing;
+        }
+        break;
+      case Axis.horizontal:
+        if (childSize.width + crossAxisSpacing > model.last.maxSliver) {
+          model.last.maxSliver = childSize.width + crossAxisSpacing;
+        }
+        break;
+    }
+
+    return DynamicSliverGridGeometry(
+      scrollOffset: scrollOffset,
+      crossAxisOffset: currentSizeUsed,
+      mainAxisExtent: childSize.height,
+      crossAxisExtent: childSize.width,
+    );
+  }
+
+  @override
+  bool reachedTargetScrollOffset(double targetOffset) {
+    if (model.last.scrollOffset > targetOffset) {
+      return true;
+    }
+    return false;
```

Comment:
No need for the if here:

```suggestion
    return model.last.scrollOffset > targetOffset;
```